src/lib/Event/Subscriber/RecommendationEventSubscriber.php

```diff
@@ -88,15 +83,19 @@ public function onRecommendationResponse(RecommendationResponseEvent $event): vo
      */
     private function getRecommendationRequest(ParameterBag $parameterBag): RecommendationRequest
     {
-        $contextItems = (int) $parameterBag->get(Request::CONTEXT_ITEMS_KEY, 0);
+        $contextItem = null;
+        $content = $parameterBag->get(Request::CONTEXT_ITEMS_KEY);
+        if ($content instanceof Content) {
+            $contextItem = $this->repositoryConfigResolver->useRemoteId() ? $content->contentInfo->remoteId : $content->id;
```

Comment:
as said in #106 this is not the cleanest approach. This should be separated in a dedicated way so consumer knows what type of data to expect.

Reply:
As I answer here: https://github.com/ezsystems/ezrecommendation-client/pull/106#discussion_r807997222 content is identified in reco engine by contentId.